ion/tokenizer.go

```diff
@@ -1403,6 +1403,10 @@ func getLongStringCommentsHandler(t *tokenizer, isClob bool) commentHandler {
 	return t.skipCommentsHandler
 }
 
+// If isClob is false, we write the UTF-8 encoding of rune, into the string builder.
+// But when isClob is true, we have a text clob. In this case, we write the bytes of
+// current rune. Clob cannot hold non 7-bit ASCII characters and to avoid truncating
+// a rune beyond 127, we store the bytes in the string builder.
```

Comment:
Making a note for future cleanup: I think that this code path is an instance of issue #48. We're using a string builder as a byte buffer; this works but is pretty confusing for readers. The end result isn't required to be a valid string.

> In this case, we write the bytes of current rune.

This is a bit muddy to me; when `isClob` is `true`, the current rune, `c`, is guaranteed to be a single byte of an unknown encoding.

> Clob cannot hold non 7-bit ASCII characters and to avoid truncating a rune beyond 127, we store the bytes in the string builder.

The text clob we're reading from can only hold 7-bit ASCII characters, but at this point in the code that's no longer relevant. We've already read the next byte (escaped or otherwise) from that clob; someone reading this function's documentation will be concerned with the contents of `c`, not its original format.

```go
// If `isClob` is false, we write the UTF-8 encoding of `c` into the string builder. When 
// `writeCharToStringBuilder` returns, `sb` will contain a valid string.
// If `isClob` is true, `c` is a single byte of an unknown encoding. We will write that byte to 
// the string builder as-is. When `writeCharToStringBuilder` returns, the contents of `sb` 
// must be treated as a byte array of unknown encoding.
```

Now that I've spent some more time with this, I do think we should refactor it. The cognitive burden is pretty high. It doesn't need to block this merge, though. I'll open a ticket to track it.